### DISCUSSION TITLE
Extracting Faces during simulation

### DISCUSSION THREAD

**Discussion by Nick20500** [google-deepmind/mujoco#2224] [Asking for Help]
### Intro

Hi!

I am a undergrad student at ETH Zurich, I use MuJoCo for my research on Multimaterial deformable objects.

### My setup

I use MuJoCo 3.2.5 with python bindings.

### My question

In my simulations I use flexcomp elements where I load .msh files into them. Everything works fine.
To generate a dataset of the objects deforming over time, I don't just want to save pixel frames but also mesh/object files. 
So far, I am extracting the vertice information for the object in the simulation with:
`sim_points = mj_data.flexvert_xpos.reshape(-1, 3)`
For viewing the results I also would wan't to extract the faces of the triangles or tetrahedra (which ever mujoco uses during simulation).

I am pretty sure that something should exist, since the renderer which generates pixel images is also able to render faces and not only the vertices but I haven't found anything in the documentation.

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by quagla** [comment]
Is this what you are asking? https://github.com/google-deepmind/mujoco/blob/main/include/mujoco/mjmodel.h#L874
i.e. you can use `model.flex_elem`.
